### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by heyhippari** [jellyfin/jellyfin-vue#278] [General]
This is the expected roadmap towards a release of the Vue client. This is by no means exhaustive, but it should help users and contributors to get a feel for what is deemed as necessary before a new version is published.

This list is not final and is subject to change.

## Preview Release 1 - "FANCY"

[Github Milestone](https://github.com/jellyfin/jellyfin-vue/milestone/1)

*This release will be the first official public showing of the Vue client. It's expected to have the minimum required for using the client, while still needing the old web client for most tasks.*

* [x] Refactor the authentication (#263)
* [x] Refactor the library view (#267)
* [x] Move the video player to a modal (#290)
* [x] Implement a basic playback manager (#290)
* [x] Implement websocket support (#125)
* [x] Implement a separate server selection page (#122)
* [x] Implement public users on the login page (#293)
* [x] Implement details pages for music (#264)
* [x] Implement details pages for TV episodes (#556)
* [x] Implement types in library pages (Show by Album, Artist, Collections, Actors, etc) (#267)
* [x] Implement a basic video player OSD (#477)
* [x] Implement resuming in the playback manager (#605)
* [x] Implement a basic UI for the audio player & hook to the playback manager (#399)
* [x] Implement a custom theme (#391)
* [ ] Implement basic search (#345)
* [x] Refactor the home sections to use Swiper (#412)
* [ ] Refactor the home sections to use different components
* [x] Implement the startup wizard (#284)
* [x] Implement library scanning (Basics done in #519, enough for release 1)
* [x] Implement a full screen visualization of the current queue (#520)
* [ ] Ensure everything is fully responsive

## Preview Release 2 - "Voice"

*This release will start to polish the existing features further, while adding music support.*

* [x] Implement an audio queue editor (#609)
* [ ] Implement a Howler backend for the music player

*More to be added as we reach the end of Preview Release 1.*

## Preview Release 3 - "Peek-A-Boo"

*This release will add support for images and books.*

* [ ] Implement an image viewer
* [ ] Implement a PDF player
* [ ] Implement a CBR/CBZ player
* [ ] Implement an EPUB player

*More to be added as we reach the end of Preview Release 2.*

## Preview Release 4 - "Deja Vu"

*This release will focus on settings, both for the user and the administrator.*

*More to be added as we reach the end of Preview Release 3.*

## Preview Release 5 - "Night Aviation"

*This release will focus on client customization (Themes and UI/UX settings).*

*More to be added as we reach the end of Preview Release 4.*

## Preview Release 6 - "Black Pearl"

*This release will focus on Live TV.*

*More to be added as we reach the end of Preview Release 5.*

**Comment by Bitwolfies** [comment]
These stage names gave me a nice laugh. :)

**Reply by heyhippari** [reply]
I felt like preview releases needed codenames to be true releases :smile:

So I did the usual strategy that lazy devs take and used some things that I like... I'll let the community try to figure it out :wink:
